java-frontend/src/test/java/org/sonar/java/model/expression/InstanceOfTreeImplTest.java

```diff
@@ -61,18 +60,12 @@ void test_PatternInstanceOfTree() {
 
   @Test
   void test_GuardedPatternInstanceOfTree() {
-    InstanceOfTreeImpl ioti = instanceOf("o instanceof (String s && s.length() > 10)");
-    assertThat(ioti.is(Tree.Kind.PATTERN_INSTANCE_OF)).isTrue();
-
-    PatternInstanceOfTree piot = ioti;
-    assertThat(piot.expression()).isNotNull();
-    assertThat(piot.instanceofKeyword()).isNotNull();
-    VariableTree variable = piot.variable();
-    assertThat(variable).isNotNull();
-    assertThat(variable.simpleName().name()).isEqualTo("s");
-    assertThat(variable.type().symbolType().is("java.lang.String")).isTrue();
-    // instanceof with guarded pattern is not fully parsed by ECJ in this version.
-    // It does not crash though, and as it is a preview feature anyway, it is fine to not support it completely.
+    ExpressionTree condition = ifCondition("o instanceof (String s && s.length() > 10)");
+    // ECJ drop the parenthesis and consider it as the two operands of a &&, while this compiles
+    assertThat(condition.is(Tree.Kind.CONDITIONAL_AND)).isTrue();
+    BinaryExpressionTree binaryExpression = (BinaryExpressionTree) condition;
+    assertThat(binaryExpression.leftOperand().is(Tree.Kind.PATTERN_INSTANCE_OF)).isTrue();
+    assertThat(binaryExpression.rightOperand().is(Tree.Kind.GREATER_THAN)).isTrue();
```

Comment:
IMO, you should not remove the previous assertions that test the type of the `s` symbol:
```
    PatternInstanceOfTree piot = (PatternInstanceOfTree) binaryExpression.leftOperand();
    assertThat(piot.expression()).isNotNull();
    assertThat(piot.instanceofKeyword()).isNotNull();
    VariableTree variable = piot.variable();
    assertThat(variable).isNotNull();
    assertThat(variable.simpleName().name()).isEqualTo("s");
    assertThat(variable.type().symbolType().is("java.lang.String")).isTrue();
```